### PR TITLE
Address project load speed regression

### LINKED ISSUE #1363: Extremely long project loading time
I have pretty large qgs file (7 MB) generated by QField plugin in QGis. The file is large, because I am using multiple styles and themes in the project and I have about 50 layers.

In previous versions project loading time was about 10 seconds on my phone. But since 1.7 version loading time is over two minutes!!

### Reproduction Steps:

1. Select a large .ggs file to be open (~7 MB)

### Expected behavior:

It should load in no more than 20 s.

### Observed behavior:

Loading time is several minutes.

**QField version:** 1.7.3 (code v1.7.3)]

## Additional information:

* Problem started happening recently, didn't happen in an older version of QField: Yes
* Problem can be reliably reproduced, doesn't happen randomly: Yes
* Problem happens with all files and projects, not only some files or projects: No

### PR DESCRIPTION
**Pull request by nirvn**
Very likely fixes #1363 , thanks hotspot.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-9dc57ca83de37fb9fbd72cd57ecaba309c944a5a-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-9dc57ca83de37fb9fbd72cd57ecaba309c944a5a-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-9dc57ca83de37fb9fbd72cd57ecaba309c944a5a-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-9dc57ca83de37fb9fbd72cd57ecaba309c944a5a-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-7ec704711cb5ae14be54b304407dbd02e91318d0-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-7ec704711cb5ae14be54b304407dbd02e91318d0-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-7ec704711cb5ae14be54b304407dbd02e91318d0-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-7ec704711cb5ae14be54b304407dbd02e91318d0-x86_64.apk)

File: src/core/layertreemodel.h [hunk 1]
```diff
@@ -100,6 +105,8 @@ class FlatLayerTreeModel : public QAbstractProxyModel
     QString mMapTheme;
     QgsProject *mProject = nullptr;
     QList<QgsLayerTreeLayer *> mLayersInTracking;
+
+    bool mFrozen;
```
Comment (m-kuhn) [inline src/core/layertreemodel.h:109]: Initialize to ... false?

**Comment by m-kuhn** [comment]
Very cool !

File: src/core/layertreemodel.h (hunk 1 above)
Comment (nirvn) [reply src/core/layertreemodel.h]: Done

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-7c234c78c9b2bd7c290a913c83a4b7bffa7ca667-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-7c234c78c9b2bd7c290a913c83a4b7bffa7ca667-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-7c234c78c9b2bd7c290a913c83a4b7bffa7ca667-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1364-7c234c78c9b2bd7c290a913c83a4b7bffa7ca667-x86_64.apk)

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**